### PR TITLE
[Feature] Add mutation policy acceptance tests

### PR DESCRIPTION
**Pull request by ghost**
1. **What this PR does / why we need it**:

2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

3. **Additional information**
Related task:
[TMC-35638](https://jira.eng.vmware.com/browse/TMC-35638)

4. **Special notes for your reviewer**

### CODE REVIEW

**Review by ramya-bangera — COMMENTED** [review]
LGTM

Can you also add mock tests ? you can refer to gitops/package install  resource to see how the mock tests work

File: examples/resources/mutation_policy/resource_cluster_scoped_label_mutation_policy.tf [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      }
+    }
+    namespace_selector {
+      match_expressions = [
+        {
+          key      = "component"
+          operator = "NotIn"
+          values   = [
+            "api-server",
+            "agent-gateway"
```
Comment (ausafaq) [inline examples/resources/mutation_policy/resource_cluster_scoped_label_mutation_policy.tf:37]: just curious, are `api-server` and `agent-gateway` valid values for a namespace?

File: internal/models/policy/recipe/mutation/common/scope.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package policyrecipemutationcommonmodel
 
 const (
-	Asterisk   VmwareTanzuManageV1alpha1CommonPolicySpecMutationV1Scope = "All"
+	Asterisk   VmwareTanzuManageV1alpha1CommonPolicySpecMutationV1Scope = "*"
```
[the remark this answers is not in the document]
Comment (ausafaq) [reply internal/models/policy/recipe/mutation/common/scope.go:4]: nit: can we rename `Asterisk` to `All` if possible?

**Comment by ausafaq** [comment]
LGTM. Thanks @ramya-bangera and @ekoop-terasky!

File: examples/resources/mutation_policy/resource_cluster_scoped_label_mutation_policy.tf (hunk 1 above)
Comment (ekoop-terasky) [reply examples/resources/mutation_policy/resource_cluster_scoped_label_mutation_policy.tf:37]: I would say yes. Reason for that conclusion is that namespace quota policy, security policy and image policy example *.tf files have same **namespace_selector** objects with **match_expressions** sub objects containing `api-server` and `agent-gateway`.

File: internal/models/policy/recipe/mutation/common/scope.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ekoop-terasky) [reply internal/models/policy/recipe/mutation/common/scope.go]: @ausafaq Sure. I pushed the change.

**Pull request merged**